src/main/scala/analysis/EdgeFunctionLattice.scala

```diff
@@ -62,16 +62,38 @@ class EdgeFunctionLattice[T, L <: Lattice[T]](val valuelattice: L) extends Latti
     def composeWith(e: EdgeFunction[T]): EdgeFunction[T] = this
 
     def joinWith(e: EdgeFunction[T]): EdgeFunction[T] =
-      if (e == this || c == valuelattice.top) this
-      else if (c == valuelattice.bottom) e
```

Comment:
Why remove this?

Reply:
These cases will be accounted for by the join edge, but removing this means we don't need to do call top (which isn't defined on some lattices such as the powerset lattice)